src/Psalm/Internal/Analyzer/Statements/Expression/BinaryOp/ArithmeticOpAnalyzer.php

```diff
@@ -867,4 +893,134 @@ public static function arithmeticOperation(
 
         return $calculated_type;
     }
+
+    private static function analyzeOperandsBetweenIntRange(
+        PhpParser\Node $parent,
+        ?Type\Union &$result_type,
+        TIntRange $left_type_part,
+        TIntRange $right_type_part
+    ): void {
+        if ($parent instanceof PhpParser\Node\Expr\BinaryOp\Div) {
+            //can't assume an int range will stay int after division
+            if (!$result_type) {
+                $result_type = new Type\Union([new Type\Atomic\TInt(), new Type\Atomic\TFloat()]);
+            } else {
+                $result_type = Type::combineUnionTypes(
+                    new Type\Union([new Type\Atomic\TInt(), new Type\Atomic\TFloat()]),
+                    $result_type
+                );
+            }
```

Comment:
Potential future refactoring: allowing `null` for `Type::combineUnionTypes()` arguments would make these `if () else`s easier to read:
```php
            if (!$result_type) {
                $result_type = new Type\Union([new Type\Atomic\TInt(), new Type\Atomic\TFloat()]);
            } else {
                $result_type = Type::combineUnionTypes(
                    new Type\Union([new Type\Atomic\TInt(), new Type\Atomic\TFloat()]),
                    $result_type
                );
            }

// c.f.

            $result_type = Type::combineUnionTypes(
                new Type\Union([new Type\Atomic\TInt(), new Type\Atomic\TFloat()]),
                $result_type
            );         
```

Alternatively we could have a new method with that semantic.

Reply:
or we may coalesce with an empty Union and check if it resolve as intended...

Maybe create a new issue on that, I'd prefer avoiding adding this to this PR, it's already big enough